src/pages/workspaces/workspace/Data.test.ts

```diff
@@ -0,0 +1,137 @@
+
+const { WorkspaceData } = testOnlyExports;
+
+type WorkspaceContainerExports = typeof import('src/pages/workspaces/workspace/WorkspaceContainer');
+jest.mock('src/pages/workspaces/workspace/WorkspaceContainer', (): WorkspaceContainerExports => {
+  return {
+    ...jest.requireActual<WorkspaceContainerExports>('src/pages/workspaces/workspace/WorkspaceContainer'),
+    wrapWorkspace: jest.fn().mockImplementation((_opts) => (wrappedComponent) => wrappedComponent),
+  };
+});
+
+type AjaxExports = typeof import('src/libs/ajax');
+jest.mock('src/libs/ajax', (): AjaxExports => {
+  return {
+    ...jest.requireActual<AjaxExports>('src/libs/ajax'),
+    Ajax: jest.fn(),
+  };
+});
+
+describe('WorkspaceData', () => {
+  // When Data.js is broken apart and the WorkspaceData component is converted to TypeScript,
+  // this type belongs there.
+  interface WorkspaceDataProps {
+    namespace: string;
+    name: string;
+    workspace: WorkspaceWrapper;
+    refreshWorkspace: () => void;
+    storageDetails: StorageDetails;
+  }
+
+  // SIFERS setup, see: https://medium.com/@kolodny/testing-with-sifers-c9d6bb5b362
+  function setup({
+    namespace = 'test-namespace',
+    name = 'test-name',
+    workspace,
+    refreshWorkspace = () => {},
+    storageDetails = {},
+    status = 'RUNNING' as LeoAppStatus,
+  }) {
+    type AjaxContract = ReturnType<typeof Ajax>;
+    type WorkspacesContract = AjaxContract['Workspaces'];
+    type AppsContract = AjaxContract['Apps'];
+    type WorkspaceDataContract = AjaxContract['WorkspaceData'];
+
+    const wdsApp: DeepPartial<ListAppResponse> = {
+      proxyUrls: {
+        wds: 'https://fake.wds.url/',
+      },
+      appType: 'WDS',
+    };
+
+    const mockGetSchema = jest.fn().mockResolvedValue([]);
+    const mockAjax: DeepPartial<AjaxContract> = {
+      Workspaces: {
+        workspace: (_namespace, _name) => ({
+          details: jest.fn().mockResolvedValue(workspace),
+          listSnapshots: jest.fn().mockResolvedValue([]),
+          entityMetadata: jest.fn().mockResolvedValue({}),
+        }),
+      } as DeepPartial<WorkspacesContract>,
+      WorkspaceData: {
+        getSchema: mockGetSchema,
+      } as DeepPartial<WorkspaceDataContract>,
+      Apps: {
+        listAppsV2: jest.fn().mockResolvedValue([{ ...wdsApp, status }]),
+      } as DeepPartial<AppsContract>,
+    };
+
+    asMockedFn(Ajax).mockImplementation(() => mockAjax as AjaxContract);
+
+    const workspaceDataProps = {
+      namespace,
+      name,
+      workspace,
+      refreshWorkspace,
+      storageDetails,
+    } as WorkspaceDataProps;
+
+    return { workspaceDataProps, mockGetSchema };
+  }
+
+  it('displays a waiting message for an azure workspace that is still provisioning in WDS', async () => {
+    // Arrange
+    const { workspaceDataProps } = setup({ workspace: defaultAzureWorkspace, status: 'PROVISIONING' });
+
+    // Act
+    await act(() => {
+      render(h(WorkspaceData, workspaceDataProps));
+    });
```

Comment:
I'm surprised nothing complains about this. `act` is typed as returning a Promise only if its given a function that returns a Promise.

It seems like this suggestion is unnecessary, but it's in line with existing code.

```suggestion
    await act(async () => {
      render(h(WorkspaceData, workspaceDataProps));
    });
```

Reply:
Hah...I don't really grok `async` / `await` yet, but was trying to get the tersest invocation possible and the test suite and linter didn't yell at me for stripping these out.  I'll add this back in though for consistency though.